### PR TITLE
Add iterative code for tower of hanoi in C

### PR DESCRIPTION
**Pull request by GuptaSamridhi**
**Changes:**

The original code was recursive so added a fast iterative code in C.

### CODE REVIEW

File: code/mathematical_algorithms/src/tower_of_hanoi/tower_of_hanoi_iterative.c [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#include <stdio.h>
+#include <math.h>
+
+void toh (int n) {
```
Comment (YashovardhanSiramdas) [inline code/mathematical_algorithms/src/tower_of_hanoi/tower_of_hanoi_iterative.c]: Function return type and Function name should be in separate lines.

File: code/mathematical_algorithms/src/tower_of_hanoi/tower_of_hanoi_iterative.c [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    int steps, start, end;
+    steps = pow (2, n);    // Minimum number of moves required
+    for (int i = 1; i < steps; i ++) {
```
Comment (YashovardhanSiramdas) [inline code/mathematical_algorithms/src/tower_of_hanoi/tower_of_hanoi_iterative.c]: Declare looping variables above the loop for compatibility with older C compilers

File: code/mathematical_algorithms/src/tower_of_hanoi/tower_of_hanoi_iterative.c [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    printf ("Disc %d shifted from %d to %d\n", i, start + 1, end + 1);
+    }
+}
+
+int main() {
+    int n;
+    printf ("Enter the number of discs : ");
+    scanf ("%d", &n);
+    toh(n);
+    return 0;
```
Comment (YashovardhanSiramdas) [inline code/mathematical_algorithms/src/tower_of_hanoi/tower_of_hanoi_iterative.c]: ```return``` values should be wrapped in parentheses

**Comment by GuptaSamridhi** [comment]
I have addressed the comments in my next commit.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**